endpoints/openrtb2/auction.go

```diff
@@ -133,8 +123,17 @@ func (deps *endpointDeps) Auction(w http.ResponseWriter, r *http.Request, _ http
 	defer cancel()
 
 	usersyncs := usersync.ParsePBSCookieFromRequest(r, &(deps.cfg.HostCookie))
+	if req.Site != nil && req.Site.Publisher != nil && req.Site.Publisher.ID != "" {
+		labels.PubID = req.Site.Publisher.ID
+	} else {
+		labels.PubID = "UNKNOWN"
```

Comment:
Why are we setting `labels.PubID = "UNKNOWN"` over here? My suggestion is that in the if block on L131, we check if `req.app.publisher.id != ""` and set it to that if it's true and in the else block check if `req.site.publisher.id != ""` and set it to that if it's true and then after the if/else block check if `labels.PubID = ""` and if it is then set it to `UNKNOWN`. According to the current code, if it's an app request then you're setting it to `UNKNOWN` first always and then resetting it to what `req.app.publisher.id` has.

Same comment goes for `video_auction.go` too

Reply:
I didn't quite understand Mansi, I'm sorry

Reply:
Do you mean something like?
**BEFORE**
```
 125     usersyncs := usersync.ParsePBSCookieFromRequest(r, &(deps.cfg.HostCookie))
 126     if req.Site != nil && req.Site.Publisher != nil && req.Site.Publisher.ID != "" {
 127         labels.PubID = req.Site.Publisher.ID
 128     } else {
 129         labels.PubID = "UNKNOWN"
 130     }
 131     if req.App != nil {
 132         labels.Source = pbsmetrics.DemandApp
 133         labels.RType = pbsmetrics.ReqTypeORTB2App
 134         if labels.PubID == "UNKNOWN" && req.App.Publisher != nil && req.App.Publisher.ID != "" {
 135             labels.PubID = req.App.Publisher.ID
 136         }
 137     } else {
 138         labels.Source = pbsmetrics.DemandWeb
 139         if usersyncs.LiveSyncCount() == 0 {
 140             labels.CookieFlag = pbsmetrics.CookieFlagNo
 141         } else {
 142             labels.CookieFlag = pbsmetrics.CookieFlagYes
 143         }
 144     }
auction.go
```
**AFTER**
```
 125     usersyncs := usersync.ParsePBSCookieFromRequest(r, &(deps.cfg.HostCookie))
 126     if req.Site != nil && req.Site.Publisher != nil && req.Site.Publisher.ID != "" {
 127         labels.PubID = req.Site.Publisher.ID
 128     }
 129     if req.App != nil {
 130         labels.Source = pbsmetrics.DemandApp
 131         labels.RType = pbsmetrics.ReqTypeORTB2App
 132         if req.App.Publisher != nil && req.App.Publisher.ID != "" {
 133             labels.PubID = req.App.Publisher.ID
 134         } else {
 135             labels.PubID = "UNKNOWN"
 136         }
 137     } else {
 138         labels.Source = pbsmetrics.DemandWeb
 139         if usersyncs.LiveSyncCount() == 0 {
 140             labels.CookieFlag = pbsmetrics.CookieFlagNo
 141         } else {
 142             labels.CookieFlag = pbsmetrics.CookieFlagYes
 143         }
 144     }
auction.go [+]
```

Reply:
But apparently a request will have one value or the other (Site or App) so if we find a site publisher value in the if statement in line 126, we'll most likely won't find a value for App.Publisher latter. What do you think?

Reply:
```
	if req.App != nil {
		labels.Source = pbsmetrics.DemandApp
		labels.RType = pbsmetrics.ReqTypeORTB2App
		if req.App.Publisher != nil && req.App.Publisher.ID != "" {
			labels.PubID = req.App.Publisher.ID
		}
	} else {
		labels.Source = pbsmetrics.DemandWeb
		if usersyncs.LiveSyncCount() == 0 {
			labels.CookieFlag = pbsmetrics.CookieFlagNo
		} else {
			labels.CookieFlag = pbsmetrics.CookieFlagYes
		}
                if req.Site != nil && req.Site.Publisher != nil && req.Site.Publisher.ID != "" {
		        labels.PubID = req.Site.Publisher.ID
	        } 
	}
        if labels.PubID == "" {
                labels.PubID = "UNKNOWN"
       }
```

That's what I am suggesting ^^

Reply:
A request is not guaranteed to have either `app` or `site`. It is an error to have both,but having neither is not strictly an error.

Reply:
@hhhjort do you want me to implement an error check to catch an instance were we have both?

Reply:
If not, errors here have been corrected and I we can probably merge (if I didn't miss any further issues that is)

Reply:
The error check should already exist elsewhere, but I am not 100% sure of it. Just pointing out you should be able to assume not both exist, but you can't assume at least one exists when constructing your logic.